src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -422,12 +421,6 @@ QueryPipeline InterpreterInsertQuery::addInsertToSelectPipeline(ASTInsertQuery &
         return std::make_shared<ExpressionTransform>(in_header, actions);
     });
 
-    /// We need to convert Sparse columns to full if the destination storage doesn't support them.
-    pipeline.addSimpleTransform([&](const Block & in_header) -> ProcessorPtr
-    {
-        return std::make_shared<MaterializingTransform>(in_header, !table->supportsSparseSerialization());
```

Comment:
Dont we lose `convertToFullColumnIfConst` which was a part of `MaterializingTransform` here?

Reply:
This `MaterializingTransform` added only for sparse columns, so I would say - no, but let's see.